aws/credentials/ssocreds/provider.go

```diff
@@ -54,6 +54,19 @@ type Provider struct {
 
 	// The URL that points to the organization's AWS Single Sign-On (AWS SSO) user portal.
 	StartURL string
+
+	// The filepath the cached token will be retrieved from. If unset Provider will
+	// use the startURL to determine the filepath at.
+	//
+	//    ~/.aws/sso/cache/<sha1-hex-encoded-startURL>.json
+	//
+	// If custom cached token filepath is used, the Provider's startUrl
+	// parameter will be ignored.
+	CachedTokenFilepath string
+
+	// Used by the SSOCredentialProvider if a token configuration
+	// profile is used in the shared config
+	SSOTokenProvider *SSOTokenProvider
```

Comment:
@aajtodd @lucix-aws i am curious about your opinion here. 

in my [V2 implementation ](https://github.com/aws/aws-sdk-go-v2/blob/main/credentials/ssocreds/sso_credentials_provider.go#L51) i did this same thing. 

at the time, not knowing Go that well (or the Go SDK), this seemed fine to me. but i think i prob should have used the `TokenProvider` interface rather than the `SSOTokenProvider` struct implementation. this is because our use of it doesnt seem to depend on anything specific to the `SSOTokenProvider`: its just calling `RetrieveBearerToken`. also, another benefit as @wty-Bryant  pointed out in a conversation I had with him, if this were an interface we can more easily unit test it.

Reply:
At first glance I think that's ok if you want to go down that route.

Reply:
Ok I will try to implement the interface route